### PR TITLE
keyguard: init at 1.6.0

### LINKED ISSUE #333201: Package request: keyguard
**Project description**

Keyguard is an alternative client for the [Bitwarden® platform](https://bitwarden.com/), created to provide the best user experience possible.

**Metadata**

* homepage URL: https://github.com/AChep/keyguard-app
* source URL: https://github.com/AChep/keyguard-app
* license: unfree
* platforms: linux, darwin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by ghost**
## Description of changes
- Init keyguard at 1.6.0
- Fixes https://github.com/NixOS/nixpkgs/issues/333201

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -2041,6 +2041,12 @@
     github = "auchter";
     githubId = 1190483;
   };
+  aucub = {
```
Comment (sikmir) [inline maintainers/maintainer-list.nix:2044]: Please add yourself to maintainer-list in a separate commit.

File: pkgs/by-name/ke/keyguard/package.nix [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  };
+in
+stdenv.mkDerivation (finalAttrs: {
+  inherit
+    pname
+    version
+    src
+    gradleBuildTask
+    gradleUpdateTask
+    ;
```
Comment (SuperSandro2000) [inline pkgs/by-name/ke/keyguard/package.nix:40]:
```suggestion
}:
let
  gradleBuildTask = ":desktopApp:packageDeb";
  gradleUpdateTask = gradleBuildTask;

in
stdenv.mkDerivation (finalAttrs: {
  pname = "keyguard";
  version = "1.6.0";

  src = fetchFromGitHub {
    owner = "AChep";
    repo = "keyguard-app";
    rev = "16e02dee823184058d43a706ed6a34d03e29680b";
    hash = "sha256-jiQnLMGUdM2UcW4c/zz2XqEVix5B68wEvpPtVAnVfEI=";
  };

  inherit gradleBuildTask gradleUpdateTask;
```

File: pkgs/by-name/ke/keyguard/package.nix [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    useBwrap = false;
+  };
+
+  doCheck = false;
+
+  __darwinAllowLocalNetworking = true;
+
+  gradleFlags = [ "-Dorg.gradle.java.home=${jdk17}" ];
+
+  env.JAVA_HOME = "${jdk17}";
```
Comment (SuperSandro2000) [inline pkgs/by-name/ke/keyguard/package.nix:65]:
```suggestion
  env.JAVA_HOME = jdk17;
```

File: pkgs/by-name/ke/keyguard/package.nix [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir $out
+    dpkg -x ./desktopApp/build/compose/binaries/main/deb/${pname}_*.deb $out
+    mv $out/opt/${pname}/* -t $out/
+    mkdir -p $out/share/applications/
+    install -Dm0644 $out/lib/${pname}-Keyguard.desktop $out/share/applications/Keyguard.desktop
```
Comment (SuperSandro2000) [inline pkgs/by-name/ke/keyguard/package.nix:86]:
```suggestion
    dpkg -x ./desktopApp/build/compose/binaries/main/deb/keyguard_*.deb $out
    mv $out/opt/keyguard/* -t $out/
    install -Dm0644 $out/lib/keyguard-Keyguard.desktop $out/share/applications/Keyguard.desktop
```

File: pkgs/by-name/ke/keyguard/package.nix [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    glib
+    cups
+    lcms2
+    alsa-lib
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir $out
```
Comment (SuperSandro2000) [inline pkgs/by-name/ke/keyguard/package.nix:82]:
```suggestion
    mkdir -p $out/share/applications/
```

File: pkgs/by-name/ke/keyguard/package.nix [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+  lcms2,
+  alsa-lib,
+}:
+let
+  gradleBuildTask = ":desktopApp:packageDeb";
+  gradleUpdateTask = gradleBuildTask;
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "keyguard";
+
```
Comment (SuperSandro2000) [inline pkgs/by-name/ke/keyguard/package.nix:27]:
```suggestion
```

**Pull request merged**